### PR TITLE
feat(lint): lint by path

### LINKED ISSUE #311: linter should accept a chart path

### PR DESCRIPTION
**Pull request by lanej**
* accepts reference to chart in home directory or path
* remove useless flag test
* Fixes #311

### CODE REVIEW

File: util/helm.go [hunk 1]
```diff
@@ -122,3 +122,8 @@ func CopyDir(src, dst string) error {
 	filepath.Walk(src, walker)
 	return failure
 }
+
+// ExpandChartPath - Expand chart path from name and home directory
+func ExpandChartPath(dir, chartName string) string {
```
Comment (adamreese) [inline util/helm.go]: I would call this `WorkspaceChartPath(...)` so we can add a `CacheChartPath(...)` or `RepoChartPath(...)`

Comment (lanej) [inline util/helm.go]: wisel

Comment (technosophos) [inline util/helm.go]: I was just thinking last night that I really wanted a library of functions to do this for me. My code is totally littered with filepath.Join calls for the same 4 things: base directory (where config.yaml is), workspace charts, charts in a repo, and directory where repos are.

Comment (adamreese) [inline util/helm.go]: I was just looking into that.  We also have the path const defined `WorkspacePath`, `WorkspaceChartPath`, etc which do not respect the config setting

Comment (lanej) [inline util/helm.go]: I rabbit-holed on the directory structure functions.  Just making it a better name for now.  I'll work on the directory refactor next.

**Comment by adamreese** [comment]
```
% ./bin/helm lint ~/.helm/workspace/charts/imnotreal                                                                                                            [e5c6e72] ruby-2.2.3
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x20 pc=0x78a02]

goroutine 1 [running]:
github.com/helm/helm/action.directoryStructure(0xc8201b0c60, 0x58, 0xc8201b0caf, 0x0, 0x0, 0x0)
	/Users/adamreese/p/go/src/github.com/helm/helm/action/lint.go:96 +0x322
github.com/helm/helm/action.Lint(0xc8201b0c60, 0x58)
	/Users/adamreese/p/go/src/github.com/helm/helm/action/lint.go:59 +0xb8
github.com/helm/helm/cli.lint(0xc8201b67e0)
	/Users/adamreese/p/go/src/github.com/helm/helm/cli/lint.go:48 +0x22e
github.com/helm/helm/vendor/github.com/codegangsta/cli.Command.Run(0x80ee50, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x87f780, 0x15, 0x0, ...)
	/Users/adamreese/p/go/src/github.com/helm/helm/vendor/github.com/codegangsta/cli/command.go:131 +0x1055
github.com/helm/helm/vendor/github.com/codegangsta/cli.(*App).Run(0xc8201b65a0, 0xc82006a120, 0x3, 0x3, 0x0, 0x0)
	/Users/adamreese/p/go/src/github.com/helm/helm/vendor/github.com/codegangsta/cli/app.go:175 +0xfd6
github.com/helm/helm/vendor/github.com/codegangsta/cli.(*App).RunAndExitOnError(0xc8201b65a0)
	/Users/adamreese/p/go/src/github.com/helm/helm/vendor/github.com/codegangsta/cli/app.go:186 +0x54
main.main()
	/Users/adamreese/p/go/src/github.com/helm/helm/helm.go:6 +0x25

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/asm_amd64.s:1721 +0x1

goroutine 19 [chan receive]:
github.com/helm/helm/vendor/github.com/golang/glog.(*loggingT).flushDaemon(0xbf5280)
	/Users/adamreese/p/go/src/github.com/helm/helm/vendor/github.com/golang/glog/glog.go:879 +0x67
created by github.com/helm/helm/vendor/github.com/golang/glog.init.1
	/Users/adamreese/p/go/src/github.com/helm/helm/vendor/github.com/golang/glog/glog.go:410 +0x297
```

**Comment by adamreese** [comment]
panics if chart path does not exist

**Comment by lanej** [comment]
good catch

**Comment by adamreese** [comment]
@lanej good to go

**Pull request merged**